### ISSUE TITLE
Rule Suggestion: disallow expression on same line as SELECT (if num_expressions>1)

### ISSUE THREAD

**Issue by aaronsteers** [sqlfluff/sqlfluff#428]
Given this example:

```sql
SELECT Md5(-1)            AS gl_account_key,
       0                  AS gl_account_number,
       'Unknown'          AS gl_account_name,
       'Unknown'          AS gl_major_category,
       'Unknown'          AS reporting_major_category
```

SQLFluff fixes as:

```sql
SELECT Md5(-1)            AS gl_account_key,
    0                  AS gl_account_number,
    'Unknown'          AS gl_account_name,
    'Unknown'          AS gl_major_category,
    'Unknown'          AS reporting_major_category
```

Whereas I think the best practice should be:

```sql
SELECT 
    Md5(-1)            AS gl_account_key,
    0                  AS gl_account_number,
    'Unknown'          AS gl_account_name,
    'Unknown'          AS gl_major_category,
    'Unknown'          AS reporting_major_category
```

In other words: `If the number of expressions is greater than one, none of them should share a line with SELECT and all should be indented the standard 4 characters.`

Thoughts? Agree/disagree?

I qualify based on the number of expressions because a case can be made that this:

```sql
SELECT
    mycol
FROM mytable
```

...is less desirable/readable than:

```sql
SELECT mycol
FROM mytable
```

**Comment by alanmcruickshank** [comment]
I'm open to this as a rule, with a few potential exceptions.

- If there is only a single column in the select, then the rule does not apply (as per your last example), this would be similar to the logic of L013, which allows `SELECT 1 FROM blah` without an alias for the column.
- Your first example actually has a hanging indent, and I think these should still be allowed, although perhaps this should be configurable for this rule. I'll add some examples below.
- The configuration from how much to indent by default should come from the master config and respect the overall settings for tabs/spaces etc...


e.g.

**GOOD**
*(Scalar query)*
```
SELECT 1
FROM foo
```
*(Hanging indent)*
```
SELECT a, b,
       c, d
FROM bar
```

*BAD*
```
SELECT a,
    b
FROM foobar
```
... should fix to ...
```
SELECT
    a,
    b
FROM foobar
```

**Comment by aaronsteers** [comment]
Yes, totally agree with you on all points. Personally, I don't prefer the hanging indent but I recognize that both can have merit. 

And yes, sounds like we're on the same page that expressions should either all be left-aligned to the (by default) four character indent - which forces the first expression onto it's own line - or else subsequent expressions should share the hanging indent start point (column position 9, I believe).

Not sure yet how this gets implemented but I'm happy to help.

**Comment by pwildenhain** [comment]
FYI this is a duplicate of #352

**Comment by aaronsteers** [comment]
@pwildenhain - Indeed. Thank you for linking the two. Where the two differ, as far as I can tell, is just the [comment](https://github.com/sqlfluff/sqlfluff/issues/352#issuecomment-631744328) from Alan on #352 related to multiple first line columns being okay, whereas here I've suggested only allowing when a single expression is named.

Taking the safe assumption that line length is already enforced, I think I'm okay with all of these:

```
select col_a, col_b from my_table
```

```
select col_a, col_b
from my_table
```

```
select col_a
from my_table
```

Whereas these would be **anti-patterns**:

```
select col_a,
    col_b
from my_table
```

```
select col_a, col_b
    col_c
from my_table
```

**Comment by NiallRees** [comment]
Resolved by #606 I believe
